pkg/controller/velerobackup/velerobackup_controller.go

```diff
@@ -169,74 +185,124 @@ func (r *ReconcileBackup) createVeleroBackupObject(namespace string) error {
+	found := true
+	if err != nil {
+		if errors.IsNotFound(err) {
+			nsLogger.Info("First time backing up this namespace, creating CheckPoint object, ")
+			cp.Name = hiveconstants.CheckpointName
+			cp.Namespace = namespace
+			found = false
+		} else {
+			nsLogger.WithError(err).Error("Failed getting CheckPoint object for namespace.")
+			return nil, false, err
 		}
 	}
 
-	return filteredHiveObjects, nil
+	return cp, found, nil
 }
 
-func (r *ReconcileBackup) updateHiveObjectsLastBackupChecksum(hobjs []*hiveObject) error {
-	for _, hobj := range hobjs {
-		hobjLogger := hobj.addLoggerFields()
+func (r *ReconcileBackup) createOrUpdateNamespaceCheckpoint(cp *hivev1.Checkpoint, found bool) error {
+	nsLogger := addNamespaceLoggerFields(r.logger, cp.Namespace)
+	var err error
 
-		hobj.setChecksumAnnotation()
-		obj := hobj.object
+	if found {
+		err = r.Update(context.TODO(), cp)
+	} else {
+		err = r.Create(context.TODO(), cp)
+	}
 
-		err := r.Update(context.TODO(), obj)
-		if err != nil {
-			hobjLogger.WithError(err).Error("Failed update.")
-			return err // If we error at all, then we should just return. The NS will be requeued in the future.
-		}
+	if err != nil {
+		nsLogger.WithError(err).Error("Failed to create or update CheckPoint object in namespace.")
+		return err
 	}
 
+	// Everything succeeded!
 	return nil
 }
 
-func addNamespaceLoggerFields(logger log.FieldLogger, namespace string) log.FieldLogger {
-	return logger.WithField("namespace", namespace)
-}
-
-func getRuntimeObjects(kclient client.Client, typesToList []runtime.Object, namespace string) ([]runtime.Object, error) {
-	nsObjects := []runtime.Object{}
-
-	for _, t := range typesToList {
-		listObj := t.DeepCopyObject()
-		if err := kclient.List(context.TODO(), listObj, client.InNamespace(namespace)); err != nil {
-			return nil, err
+// calculateChecksum computes the runtime object's hash WITHOUT the status as status is not used to determine if the object has changed.
+// We never return an error from this function as that could lead to a
+// situation where other objects change, but because one of the objects
+// checksum errored, then the other objects won't get backed up.
+// Instead for objects that error, we return a constant checksum value.
+// This function should rarely, if ever, return an error.
+func (r *ReconcileBackup) calculateObjectsChecksumWithoutStatus(objects ...runtime.Object) string {
+	checksums := []string{}
+	for _, object := range objects {
+		var meta *metav1.ObjectMeta
+		var spec interface{}
+		var checksum = ""
+		var err error
+
+		switch t := object.(type) {
+		case *hivev1.ClusterDeployment:
+			meta = &t.ObjectMeta
+			spec = &t.Spec
+		case *hivev1.SyncSet:
+			meta = &t.ObjectMeta
+			spec = &t.Spec
+		case *hivev1.DNSZone:
+			meta = &t.ObjectMeta
+			spec = &t.Spec
+		default:
+			r.logger.Infof("Unknown Type: %T", object)
+			checksum = errChecksum
```

Comment:
If we do the following here, then we do not have to carry the `checksum` variable and do not have to have the check later of whether the checksum is the error checksum.
```
checksum[i] = errChecksum   // or append if not taking my other suggestion
continue
```